### DISCUSSION TITLE
Is the viem client intentionally mutable?

### DISCUSSION THREAD

**Discussion by frolic** [wevm/viem#1328] [Question] [answered]
are viem client objects meant to be mutable? is the mutability used anywhere?

does wagmi mutate the client object’s chain or create a new client on e.g. network changes?

I am wondering if I can expect/guarantee a new client object per chain vs the chain within a client object being mutable

**Comment by jxom** [comment, marked as answer]
The client is intended to be immutable.

**Reply by frolic** [reply]
In that case, should we mark the return type as `readonly`?

**Reply by joaofranciscoguarda** [reply]
And what should I do if I want to have multiple chain support?

**Reply by tmm** [reply]
@joaofranciscoguarda what are you doing right now?

**Reply by joaofranciscoguarda** [reply]
@tmm For now, I'm just testing my app using only mumbai, but I trying to understand how could I change it dynamically in the backend. Probably it will be need to set in a different table what is the network that the user is connected and then instanciate another walletClient based on the chain they last used/setted. For now, it's not a problem for me so I'm not deep on in.

**Reply by joaofranciscoguarda** [reply]
@tmm thanks for the great job you are doing, hope viem become the best
